File: Security/src/AspDotNet4/CloudFoundrySingleSignon/Controllers/AccountController.cs
using CloudFoundrySingleSignon.Models;
using Steeltoe.Security.Authentication.CloudFoundry;
using System.Web;
using System.Web.Mvc;

namespace CloudFoundrySingleSignon.Controllers
{
    public class AccountController : Controller
    {
        public ActionResult AuthorizeSSO(string returnUrl)
        {
            // the value for provider must match the value used for AuthenticationType when configuring authentication (in Startup.Auth.cs)
            return new ChallengeResult(CloudFoundryDefaults.AuthenticationScheme, returnUrl ?? Url.Action("Secure", "Home"));
        }

        public ActionResult AccessDenied()
        {
            ViewData["Message"] = "Insufficient permissions.";
            return View();
        }

        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult LogOff()
        {
            Request.GetOwinContext().Authentication.SignOut();
            return RedirectToAction("Index", "Home");
        }
    }
}

File: Security/src/AspDotNet4/CloudFoundrySingleSignon-VB/Controllers/AccountController.vb
Imports CloudFoundrySingleSignon_VB.CloudFoundrySingleSignon.Models
Imports Steeltoe.Security.Authentication.CloudFoundry
Imports System.Web
Imports System.Web.Mvc

Namespace CloudFoundrySingleSignon.Controllers
    Public Class AccountController
        Inherits Controller

        Public Function AuthorizeSSO(ByVal returnUrl As String) As ActionResult
            Return New ChallengeResult(CloudFoundryDefaults.AuthenticationScheme, If(returnUrl, Url.Action("Secure", "Home")))
        End Function

        Public Function AccessDenied() As ActionResult
            ViewData("Message") = "Insufficient permissions."
            Return View()
        End Function

        <HttpPost>
        <ValidateAntiForgeryToken>
        Public Function LogOff() As ActionResult
            Request.GetOwinContext().Authentication.SignOut()
            Return RedirectToAction("Index", "Home")
        End Function
    End Class
End Namespace
